src/pudl_rmi/connect_deprish_to_ferc1.py

```diff
@@ -66,762 +191,380 @@ def execute(plant_parts_eia, deprish_eia, ferc1_to_eia, clobber=False):
-            same_smol = self.split_ferc1_data_on_split_cols(
-                same_smol,
-                data_col=f"{data_col}_own_frac",
-                idx_cols_ferc1=idx_cols_ferc1,
-                split_cols=split_cols,
+        # two dfs
+        dupes = connected_to_scale[dupe_mask]
+        non_dupes = connected_to_scale[~dupe_mask]
+        # If there are no duplicate records, then the following aggs will fail
+        # bc there is nothing to merge. So we're making a new df to output that
+        # is these non_dupes. If ther are dupes, we'll aggregate them!
+        if dupes.empty:
+            df_out = non_dupes
+        else:
+            logger.info(
+                f"Aggergating {len(dupes)} duplicate records "
+                f"({len(dupes)/len(connected_to_scale):.1%})")
+
+            # sum and weighted average!
+            de_duped = pudl.helpers.sum_and_weighted_average_agg(
+                df_in=dupes,
+                by=self.eia_pk,
+                sum_cols=self.get_treatment_cols('scale'),
+                wtavg_dict=self.wtavg_dict
             )
-        return same_smol
+            # add in the string columns
+            # TODO: add a test to ensure that the str-squish character doesn't
+            # show up in the original data columns
+            de_duped = de_duped.merge(
+                (
+                    dupes.groupby(self.eia_pk, as_index=False)
+                    .agg({k: str_concat for k
+                          in self.get_treatment_cols('str_concat')})
+                ),
+                on=self.eia_pk,
+                validate='1:1',
+                how='left'
+            ).pipe(pudl.helpers.convert_cols_dtypes, 'eia')
+
+            # merge back in the ppl idx columns
+            de_duped = (
+                de_duped.set_index('record_id_eia')
+                .merge(
+                    ppl,
+                    left_index=True,
+                    right_index=True,
+                    how='left',
+                    validate='m:1',
+                )
+                .reset_index()
+            )
+            # merge the non-dupes and de-duplicated records
+            # we're doing an inner merge here bc we don't want columns with
+            # partially null values
+            df_out = pd.concat([non_dupes, de_duped], join='inner')
+        return df_out
 
-    def split_ferc1_data_on_split_cols(self,
-                                       same_smol,
-                                       data_col,
-                                       idx_cols_ferc1,
-                                       split_cols,):
+    def many_merge_on_scale_part(
+            self,
+            to_scale: pd.DataFrame,
+            cols_to_keep: list,
+            ppl: pd.DataFrame) -> pd.DataFrame:
         """
-        Split larger ferc1 records porportionally by depreciation columns.
+        Merge a particular EIA plant-part list plant-part onto a dataframe.
```

Comment:
Not clear to me what this method does based on name and docstring, or the code/comments.